packages/viewer/src/components/scene-tree-table/scene-tree-table.tsx

```diff
@@ -0,0 +1,511 @@
+  InstancedTemplate,
+} from '../scene-tree/lib/templates';
+import { RowDataProvider } from '../scene-tree/scene-tree';
+import { getSceneTreeTableOffsetTop } from './lib/dom';
+
+interface StateMap {
+  columnElementPools?: WeakMap<
+    HTMLVertexSceneTreeTableColumnElement,
+    ElementPool
+  >;
+  headerInstances?: Array<InstancedTemplate<HTMLElement>>;
+
+  startIndex: number;
+  endIndex: number;
+  viewportRows: Row[];
+}
+
+@Component({
+  tag: 'vertex-scene-tree-table',
+  styleUrl: 'scene-tree-table.css',
+  shadow: true,
+})
+export class SceneTreeTable {
+  /**
+   * A reference to the scene tree to perform operations for interactions. Such
+   * as expansion, visibility and selection.
+   */
+  @Prop()
+  public tree?: HTMLVertexSceneTreeElement;
+
+  /**
+   * @internal
+   */
+  @Prop()
+  public controller?: SceneTreeController;
+
+  /**
+   * @internal
+   */
+  @Prop()
+  public rows: Row[] = [];
+
+  /**
+   * @internal
+   */
+  @Prop()
+  public totalRows = 0;
+
+  /**
+   * @internal
+   */
+  @Prop({ mutable: true })
+  public rowHeight = 24;
+
+  /**
+   * The number of offscreen rows above and below the viewport to render. Having
+   * a higher number reduces the chance of the browser not displaying a row
+   * while scrolling.
+   *
+   * This prop will be automatically populated based on the `overScanCount` prop
+   * specified in the parent `<vertex-scene-tree />` element.
+   */
+  @Prop()
+  public overScanCount = 25;
+
+  /**
+   * A callback that is invoked immediately before a row is about to rendered.
+   * This callback can return additional data that can be bound to in a
+   * template.
+   *
+   * This prop will be automatically populated based on the `rowData` prop
+   * specified in the parent `<vertex-scene-tree />` element.
+   *
+   * @example
+   *
+   * ```html
+   * <script>
+   *   const table = document.querySelector('vertex-scene-tree-table');
+   *   table.rowData = (row) => {
+   *     return { func: () => console.log('row', row.node.name) };
+   *   }
+   * </script>
+   *
+   * <vertex-scene-tree>
+   *  <vertex-scene-tree-table>
+   *    <vertex-scene-tree-table-column>
+   *      <template>
+   *        <button event:click="{{row.data.func}}">Hi</button>
+   *      </template>
+   *    </vertex-scene-tree-table-column>
+   *  </vertex-scene-tree-table>
+   * </vertex-scene-tree>
+   * ```
+   */
+  @Prop()
+  public rowData?: RowDataProvider;
+
+  /**
+   * @internal
+   */
+  @Prop({ mutable: true })
+  public layoutOffset = 0;
+
+  /**
+   * @internal
+   */
+  @Prop({ mutable: true })
+  public scrollOffset = 0;
+
+  /**
+   * @internal
+   */
+  @Prop({ mutable: true })
+  public layoutHeight?: number;
+
+  /**
+   * A flag that disables the default interactions of this component. If
+   * disabled, you can use the event handlers to be notified when certain
+   * operations are performed by the user.
+   */
+  @Prop()
+  public interactionsDisabled = false;
```

Comment:
I'd recommend keeping this out of the layout component. It's an implementation detail of the `vertex-scene-tree-table-cell` component.

Reply:
Sounds good, this was mainly for convenience in the case that all interactions for the scene tree should be disabled, but it's not much more effort to specify on a per cell template basis, so I'll remove this